system/src/system_threading.cpp

```diff
@@ -0,0 +1,84 @@
+
+
+#include "system_threading.h"
+#include "system_task.h"
+#include <time.h>
+#include <string.h>
+
+
+#if PLATFORM_THREADING
+void system_thread_idle()
+{
+    Spark_Idle_Events(true);
+}
+
+ActiveObjectThreadQueue SystemThread(ActiveObjectConfiguration(system_thread_idle, 1024*3, 100));
```

Comment:
Yep! Well caught! Fortunately the stack_size isn't used, but it did mean the system thread was waiting longer than I'd like to run the background loop.